File: src/main/java/yuuto/enhancedinventories/proxy/ProxyCommon.java
/*******************************************************************************
 * Copyright (c) 2014 Yuuto.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the GNU Public License v3.0
 * which accompanies this distribution, and is available at
 * http://www.gnu.org/licenses/gpl.html
 *
 * Contributors:
 * 	   cpw - src reference from Iron Chests
 * 	   doku/Dokucraft staff - base chest texture
 *     Yuuto - initial API and implementation
 ******************************************************************************/
package yuuto.enhancedinventories.proxy;

import net.minecraft.init.Blocks;
import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraftforge.oredict.ShapedOreRecipe;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLPostInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.network.NetworkRegistry;
import cpw.mods.fml.common.registry.GameRegistry;
import yuuto.enhancedinventories.EInventoryMaterial;
import yuuto.enhancedinventories.EnhancedInventories;
import yuuto.enhancedinventories.RecipeConverter;
import yuuto.enhancedinventories.block.BlockImprovedChestOld;
import yuuto.enhancedinventories.block.BlockLocker;
import yuuto.enhancedinventories.compat.refinedrelocation.TileImprovedSortingChest;
import yuuto.enhancedinventories.compat.refinedrelocation.TileSortingLocker;
import yuuto.enhancedinventories.gui.GuiHandler;
import yuuto.enhancedinventories.item.ItemBlockImprovedChest;
import yuuto.enhancedinventories.item.ItemBlockLocker;
import yuuto.enhancedinventories.tile.TileImprovedChestOld;
import yuuto.enhancedinventories.tile.TileLocker;
import yuuto.yuutolib.IProxy;

public class ProxyCommon implements IProxy {

	@Override
	public void preInit(FMLPreInitializationEvent event) {
		ConfigHandler.init(event);
		BlockImprovedChestOld improvedChest = EnhancedInventories.improvedChest;
		BlockLocker locker = EnhancedInventories.locker;
		GameRegistry.registerItem(EnhancedInventories.sizeUpgrade, "sizeUpgrade");
		GameRegistry.registerItem(EnhancedInventories.functionUpgrade, "functionUpgrade");
		GameRegistry.registerItem(EnhancedInventories.chestConverter, "chestConverter");
		GameRegistry.registerBlock(improvedChest, ItemBlockImprovedChest.class, "improvedChest");
		GameRegistry.registerTileEntity(TileImprovedChestOld.class, "container.ImprovedChests:ImprovedChest");
		GameRegistry.registerBlock(locker, ItemBlockLocker.class, "locker");
		GameRegistry.registerTileEntity(TileLocker.class, "container.ImprovedChests:locker");
		registerCompatBlocks();
	}
	
	public void registerCompatBlocks(){
		if(EnhancedInventories.refinedRelocation){
			GameRegistry.registerBlock(EnhancedInventories.improvedSortingChest, ItemBlockImprovedChest.class, "improvedSortingChest");
			GameRegistry.registerTileEntity(TileImprovedSortingChest.class, "container.ImprovedChests:ImprovedSortingChest");
			GameRegistry.registerBlock(EnhancedInventories.sortingLocker, ItemBlockLocker.class, "sortingLocker");
			GameRegistry.registerTileEntity(TileSortingLocker.class, "container.ImprovedChests:SortingLocker");
		}
			
	}

	@Override
	public void init(FMLInitializationEvent event) {	
		NetworkRegistry.INSTANCE.registerGuiHandler(EnhancedInventories.instance, new GuiHandler());

	}

	@Override
	public void postInit(FMLPostInitializationEvent event) {
		//System.out.println("EI Post Init");
		
	}
	
	public void registerRecipes(){
		ItemStack base = new ItemStack(EnhancedInventories.functionUpgrade, 1, 0);
		GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(EnhancedInventories.functionUpgrade, 1, 0), new Object[]{
			" c ", "cpc", " c ", 'c', "cobblestone", 'p', Items.paper 
		}));
		GameRegistry.addShapedRecipe(new ItemStack(EnhancedInventories.functionUpgrade, 1, 1), new Object[]{
			" h ", "hmh", " h ", 'h', Blocks.hopper, 'm', base
		});
		GameRegistry.addShapelessRecipe(new ItemStack(EnhancedInventories.functionUpgrade, 1, 2), 
			Blocks.tripwire_hook, base);
		for(int i = 1; i < EInventoryMaterial.values().length; i++){
			GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(EnhancedInventories.sizeUpgrade, 1, i-1), new Object[]{
				"mmm", "mbm", "mmm", 'm', EInventoryMaterial.values()[i].getMaterial(), 'b', base
			}));
			GameRegistry.addRecipe(new RecipeConverter(new ItemStack(EnhancedInventories.chestConverter, 1, i), new Object[]{
				"c","b", 
				'c', new ItemStack(EnhancedInventories.improvedChest, 1, i), 
				'b', new ItemStack(EnhancedInventories.functionUpgrade, 1, 0)
			}));
		}
	}

}

File: src/main/scala/yuuto/enhancedinventories/proxy/ProxyCommon.scala
/**
 * @author Yuuto
 */
package yuuto.enhancedinventories.proxy

import net.minecraft.block.Block
import net.minecraft.item.Item
import net.minecraftforge.common.MinecraftForge
import yuuto.enhancedinventories.EnhancedInventories
import yuuto.enhancedinventories.block.BlockAutoAssembler
import yuuto.enhancedinventories.block.BlockCabinet
import yuuto.enhancedinventories.block.BlockImprovedChest
import yuuto.enhancedinventories.block.BlockLocker
import yuuto.enhancedinventories.block.BlockPainter
import yuuto.enhancedinventories.block.BlockWorktable
import yuuto.enhancedinventories.compat.CompatLoader
import yuuto.enhancedinventories.config.EIConfiguration
import yuuto.enhancedinventories.gui.GuiHandler
import yuuto.enhancedinventories.item.ItemBlockCabinet
import yuuto.enhancedinventories.item.ItemBlockImprovedChest
import yuuto.enhancedinventories.item.ItemBlockLocker
import yuuto.enhancedinventories.item.ItemChestConverter
import yuuto.enhancedinventories.item.ItemFunctionUpgrade
import yuuto.enhancedinventories.item.ItemPaintBrush
import yuuto.enhancedinventories.item.ItemSchematic
import yuuto.enhancedinventories.item.ItemSizeUpgrade
import yuuto.enhancedinventories.tile.TileAutoAssembler
import yuuto.enhancedinventories.tile.TileCabinet
import yuuto.enhancedinventories.tile.TileImprovedChest
import yuuto.enhancedinventories.tile.TileLocker
import yuuto.enhancedinventories.tile.TilePainter
import yuuto.enhancedinventories.tile.TileWorktable
import cpw.mods.fml.common.event.FMLInitializationEvent
import cpw.mods.fml.common.event.FMLPostInitializationEvent
import cpw.mods.fml.common.event.FMLPreInitializationEvent
import cpw.mods.fml.common.network.NetworkRegistry
import cpw.mods.fml.common.registry.GameRegistry;
import yuuto.enhancedinventories.config.json.JsonRecipeFactory

object ProxyCommon{
  final val blockImprovedChest:Block = new BlockImprovedChest("improvedChest");
  final val blockLocker:Block = new BlockLocker("locker");
  final val blockCabinet:Block = new BlockCabinet("cabinet");
  final val blockWorktable:Block = new BlockWorktable("worktable");
  final val blockAutoAssembler:Block = new BlockAutoAssembler("autoAssembler");
  final val blockPainter:Block = new BlockPainter("painter");
  
  final val functionUpgrades:Item = new ItemFunctionUpgrade("upgradeFunction").setTextureNames(
      "Basic", "Hopper", "Trapped", "Security", "Reinforced"
      );
  final val sizeUpgrades:Item = new ItemSizeUpgrade("sizeUpgrade");
  final val chestConverter:Item = new ItemChestConverter("chestConverter");
  final val schematic:Item = new ItemSchematic("schematic");
  final val paintBrush:Item = new ItemPaintBrush("paintbrush");
}
class ProxyCommon {
  
  def preInit(event:FMLPreInitializationEvent) {
    EIConfiguration.init(event);
    NetworkRegistry.INSTANCE.registerGuiHandler(EnhancedInventories, new GuiHandler());
    CompatLoader.preInit();
    GameRegistry.registerBlock(ProxyCommon.blockImprovedChest, classOf[ItemBlockImprovedChest], "improvedChest");
    GameRegistry.registerBlock(ProxyCommon.blockLocker, classOf[ItemBlockLocker], "locker");
    GameRegistry.registerBlock(ProxyCommon.blockCabinet, classOf[ItemBlockCabinet], "blockCabinet");
    GameRegistry.registerBlock(ProxyCommon.blockWorktable, "blockWorktable");
    GameRegistry.registerBlock(ProxyCommon.blockAutoAssembler, "blockAutoAssembler");
    GameRegistry.registerBlock(ProxyCommon.blockPainter, "blockPainter");
    GameRegistry.registerTileEntity(classOf[TileImprovedChest], "container.ImprovedChests:ImprovedChest");
    GameRegistry.registerTileEntity(classOf[TileLocker], "container.ImprovedChests:locker");
    GameRegistry.registerTileEntity(classOf[TileCabinet], "tileCabinet");
    GameRegistry.registerTileEntity(classOf[TileWorktable], "tileWorktable");
    GameRegistry.registerTileEntity(classOf[TileAutoAssembler], "tileAutoAssembler");
    GameRegistry.registerTileEntity(classOf[TilePainter], "tilePainter");
    GameRegistry.registerItem(ProxyCommon.functionUpgrades, "functionUpgrade");
    GameRegistry.registerItem(ProxyCommon.sizeUpgrades, "sizeUpgrade");
    GameRegistry.registerItem(ProxyCommon.chestConverter, "chestConverter");
    GameRegistry.registerItem(ProxyCommon.schematic, "schematic");
    GameRegistry.registerItem(ProxyCommon.paintBrush, "paintbrush");
    //RecipeFactory.init();
  }

  def init(event:FMLInitializationEvent) {
    
  }

  def postInit(event:FMLPostInitializationEvent) {
    CompatLoader.postInit();
  }

}